dvc/command/version.py

```diff
@@ -48,9 +48,14 @@ def run(self):
             repo = Repo()
             root_directory = repo.root_dir
 
-            info += "Cache: {cache}\n".format(
-                cache=self.get_linktype_support_info(repo)
-            )
+            # cache_dir might not exist yet (e.g. after `dvc init`), and we
+            # can't auto-create it, as it might cause issues if the user
+            # later decides to enable shared cache mode with
+            # `dvc config cache.shared group`.
```

Comment:
This doesn't feel right for two reasons:
- maybe user is looking for this link type supported to configure cache and we are not showing it. There is no instruction/way to finally see that link types here, so we are simply leaving users in the dark.
- what if cache dir was created with whatever op (i.e. `dvc add`) and then user says `dvc config cache.shared group`?

Reply:
> maybe user is looking for this link type supported to configure cache and we are not showing it. There is no instruction/way to finally see that link types here, so we are simply leaving users in the dark.

True, but making `dvc version` create a directory for you is even more confusing. We can print a warning though asking to create it by-hand or `dvc add` something.

> what if cache dir was created with whatever op (i.e. dvc add) and then user says dvc config cache.shared group?

For him, it will work fine, but his colleagues will run into permission errors. Might want to add perm checks later.

Reply:
Making version create a dir is not confusing, this dir is configured up already anyway. Making `dvc version` behave differently on some arcane condition is a no-no.

So second point says that your explanation in the comment about switching to shared mode later is invalid.

Overall this patch solves no issues, while providing `dvc version` inconsistent behavior. Doesn't look good.

Reply:
@Suor making `dvc version` create something permanent is a no-no. Though, we can create and then delete it. That would be fine. Will adjust the patch.

Reply:
Though temporarily creating it as no-go too, as another user might start using it.

Reply:
Plus, it might be meant to be mounted by the user later, so we can't create a temporary directory nearby. So I would simply not create anything at all.

Reply:
Discussed with @Suor in PMs and decided to leave as-is for now.